onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h

```diff
@@ -0,0 +1,706 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "tree_ensemble_aggregator.h"
+
+namespace onnxruntime {
+namespace ml {
+
+template <typename ITYPE, typename OTYPE>
+class TreeEnsembleCommon {
+ public:
+  int64_t n_targets_or_classes_;
+
+ protected:
+  std::vector<OTYPE> base_values_;
+  POST_EVAL_TRANSFORM post_transform_;
+  AGGREGATE_FUNCTION aggregate_function_;
+  int64_t n_nodes_;
+  TreeNodeElement<OTYPE>* nodes_;
+  std::vector<TreeNodeElement<OTYPE>*> roots_;
+
+  int64_t max_tree_depth_;
+  int64_t n_trees_;
+  bool same_mode_;
+  bool has_missing_tracks_;
+  int omp_tree_;
+  int omp_N_;
+
+ public:
+  TreeEnsembleCommon(int omp_tree, int omp_N);
+  ~TreeEnsembleCommon();
+
+  void init(
+      const std::string& aggregate_function,
```

Comment:
nit: does init need to be separate or can we just do all this in the ctor? 

[https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-complete](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-complete)

Reply:
fixed